tests/integration/test_integration.py

```diff
@@ -333,6 +333,57 @@ async def test_given_vault_kv_requirer_related_when_create_secret_then_secret_is
 
         assert action_output["value"] == secret_value
 
+    @pytest.mark.abort_on_fail
+    async def test_given_vault_deployed_when_tls_access_relation_created_then_existing_certificate_replaced(
+        self, ops_test: OpsTest
+    ):
+        assert ops_test.model
+
+        vault_leader_unit = ops_test.model.units[f"{APPLICATION_NAME}/0"]
+        action = await vault_leader_unit.run("cat /var/lib/juju/storage/certs/0/ca.pem")
+        await action.wait()
+        initial_ca_cert = action.results["stdout"]
+
+        await ops_test.model.integrate(
+            relation1=f"{SELF_SIGNED_CERTIFICATES_APPLICATION_NAME}:certificates",
+            relation2=f"{APPLICATION_NAME}:tls-certificates-access",
+        )
+
+        await ops_test.model.wait_for_idle(
+            apps=[APPLICATION_NAME, SELF_SIGNED_CERTIFICATES_APPLICATION_NAME],
+            status="active",
+            timeout=1000,
+        )
+
+        action = await vault_leader_unit.run("cat /var/lib/juju/storage/certs/0/ca.pem")
+        await action.wait()
+        final_ca_cert = action.results["stdout"]
+        assert initial_ca_cert != final_ca_cert
```

Comment:
Wouldn't it be better to test that we can connect to the API using the new certificate?

Reply:
If it was not connecting, the vault unit would go back to waiting state "waiting for vault to be available" because the internal hvac library also uses the ca certificate to connect to vault to do health checks etc.

So by waiting for idle we're already checking exactly that.

Reply:
Ok, thank you!